tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1858,6 +1879,148 @@ public:
    }
 };
 
+#ifdef R__HAS_ROOT7
+void ValidateSnapshotRNTupleOutput(const RSnapshotOptions &opts, const std::string &ntupleName,
+                                   const std::string &fileName);
+
+/// Helper function to update the value of an RNTuple's field in the provided entry.
+template <typename T>
+void SetFieldsHelper(const T &value, std::string_view fieldName, ROOT::Experimental::REntry *entry)
+{
+   entry->BindValue(fieldName, std::make_shared<T>(value));
```

Comment:
Note that I think this copies all values, including vectors and other "complex" containers. Would it be possible to `BindRawPtr`? I think `SetFields` gets non-`const` references, so that should not get into our way...

Reply:
Added here:

https://github.com/root-project/root/blob/a3dbe5b83b31d449d8812ca29762a6cd3c43c298/tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx#L1893-L1898